sqlalchemy_media/stores/GoogleCloudStorge.py

```diff
@@ -0,0 +1,100 @@
+""""
+    Google cloud storage
+"""
+from io import BytesIO
+
+from google.cloud import storage
+from google.cloud.exceptions import NotFound
+
+from sqlalchemy_media.exceptions import GCPError
+from sqlalchemy_media.optionals import ensure_gcs
+from .base import Store
+from ..typing_ import FileLike
+
+
+class GoogleCloudStorge(Store):
+    """
+    Google Cloud Storage Implements Store base.
+    """
+
+    def __init__(self, bucket: str, service_account_json: str, acl: str = 'private'):
+        """
+        Initialize GoogleCloudStorge
+        :param bucket: bucket name
+        :param service_account_json: service account json file(credential)
+        :param acl: public or private
+        """
+        self.bucket = bucket
+        self._storage_client = storage.Client.from_service_account_json(service_account_json)
+        self.acl = acl
+
+    def _get_or_create_bucket(self):
+        """
+        Get bucket if exist else create a bucket
+        :return bucket object
+        """
+        ensure_gcs()
+        try:
+            return self._storage_client.get_bucket(self.bucket)
+        except NotFound:
+            return self._storage_client.create_bucket(self.bucket)
+
+    def _upload_file(self, file_name: str, data: str):
+        """
+        Create a blob and upload file,
+        add acl if needed
+        """
+        ensure_gcs()
+        try:
+            bucket = self._get_or_create_bucket()
+            blob = bucket.blob(file_name)
+            blob.upload_from_string(data)
+            if self.acl == 'public':
+                blob.make_public()
+        except Exception as e:
+            raise GCPError(e)
+
+    def locate(self, attachment) -> str:
```

Comment:
@ribrea Most of the time the resource is protected, so there won't be any public URL. I suggest using the same method as s3 store: 
```
    def locate(self, attachment) -> str:
        if self.cdn_url:
            base_url = self.cdn_url
        else:
            base_url = self.base_url
        return '%s/%s' % (base_url, attachment.path)
```

Reply:
Google's Doing the same ting
@property
    def public_url(self):
        """The public URL for this blob.

        Use :meth:`make_public` to enable anonymous access via the returned
        URL.

        :rtype: `string`
        :returns: The public URL for this blob.
        """
        return "{storage_base_url}/{bucket_name}/{quoted_name}".format(
            storage_base_url=_API_ACCESS_ENDPOINT,
            bucket_name=self.bucket.name,
            quoted_name=_quote(self.name, safe=b"/~"),
        )